linode/terraform-provider-linode
# Adding Framework validator for label attributes

## 📝 Description

Adds length and content validators for label attributes for Framework resources. Content validators are created using a new RegEx validator. All information regarding label restrictions are pulled from API code. This PR also adds PKG_NAME to `make unittest` such that we can run unit tests by folder.

## ✔️ How to Test

`make unittest PKG_NAME=linode/helper`

`make testacc`

Manual Testing:

Inside of a provider development environment (e.g. dx-devenv):

1. Add the following configuration inside of `main.tf`:


// =============================================================================
// MySQL resource
// =============================================================================
resource "linode_nodebalancer" "foobar" {
  label  = "much_much_much_much_much_too_long_of_a_label"
  region = "us-east"
  client_conn_throttle = 20
}


2. Run `terraform init -upgrade` to install the random provider.
3. Run `make apply` to apply the configuration.
5. Observe the error.
